### PR TITLE
Simplified data loader

### PR DESCRIPTION
**Pull request by theHenks**
Included `read_ldata` function to simplify data loading for a single `DataTier` and `ChannelId` including fast flattening and concatenation of tables and arrays.
The following functionality is implemented:
Example (requires a `$LEGEND_DATA_CONFIG` environment variable pointing to a legend data-config file):
    
```julia
using LegendDataManagement, LegendHDF5IO
l200 = LegendData(:l200)
filekeys = search_dsik(FileKey, l200.tier[:jldsp, :cal, :p03, :r000])

chinfo = channelinfo(l200, (:p03, :r000, :cal); system=:geds, only_processable=true)

ch = chinfo[1].channel

dsp = read_ldata(l200, :jldsp, first(filekeys), ch)
dsp = read_ldata(l200, :jldsp, :cal, :p03, :r000, ch)
dsp = read_ldata((:e_cusp, :e_trap, :blmean, :blslope), l200, :jldsp, :cal, :p03, :r000, ch)
```
`read_ldata` automitcally loads LEGEND data for a specific `DataTier` and data selection like e.g. a `FileKey` or a run-selection based for a given `ChannelId`. The `search_disk` function allows the user to search for available `DataTier` and `FileKey` on disk. The first argument can be either a selection of keys in form of a `NTuple` of `Symbol` or a [PropertyFunction](https://github.com/oschulz/PropertyFunctions.jl/tree/main) which will be applied during loading. 
It is also possible to load whole a `DataPartition` or `DataPeriod` for a given `ChannelId` ch:
```julia
dsp = read_ldata(l200, :jldsp, :cal, DataPartition(1), ch)
dsp = read_ldata(l200, :jldsp, :cal, DataPeriod(3), ch)
```
In additon, it is possible to load a random selection of `n_evts` events randomly selected from each loaded file:
```julia
dsp = read_ldata(l200, :jldsp, :cal, :p03, :r000, ch; n_evts=1000)
```
For simplicity, the ch can also be given as a `DetectorID` which will be converted internally to a `ChannelId`:
```julia
det = chinfo[1].detector
dsp = read_ldata(l200, :jldsp, :cal, :p03, :r000, det)
```

### CODE REVIEW

**Comment by oschulz** [comment]
This is still in progress, right?

**Comment by theHenks** [comment]
I need some final checks, than it is ready. But the core is in principle ready to be merged.

File: docs/src/extensions.md [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+## `LegendHDF5IO` extension
+
+LegendDataManagment provides an extension for [LegendHDF5IO](https://github.com/legend-exp/LegendHDF5IO.jl).
+This makes it possible to directly load LEGEND data from HDF5 files via the `read_ldata` function. The extension is automatically loaded when both packages are loaded. 
+Example (requires a `$LEGEND_DATA_CONFIG` environment variable pointing to a legend data-config file):
+    
+```julia
+using LegendDataManagement, LegendHDF5IO
+l200 = LegendData(:l200)
+filekeys = search_dsik(FileKey, l200.tier[:jldsp, :cal, :p03, :r000])
```
Comment (fhagemann) [inline docs/src/extensions.md:12]: Typo

**Comment by fhagemann** [comment]
@theHenks  I merged all PRs on `dev` and resolved the merge conflicts, 
meaning that it is up to date with `main` and should be good to go.

**Comment by oschulz** [comment]
Is dev ready to be merged into main then?

**Pull request merged**